src/components/shared/scanner/index.js

```diff
@@ -11,133 +13,121 @@ import CameraRoll from './cameraRoll';
 import withTheme from '../withTheme';
 import getStyles from './styles';
 
-class Scanner extends React.Component {
-  state = {
-    camera: {
-      permission: 'undetermined',
-      visible: false,
-    },
-    photo: {
-      permission: 'undetermined',
-      visible: false,
-    },
-  };
-
-  componentDidMount() {
-    this.checkPermissions();
-    AppState.addEventListener('change', this.checkPermissions);
-  }
-
-  checkPermissions = () => {
-    Permissions.checkMultiple(['camera', 'photo']).then(response => {
-      this.setPermissions(response);
-    });
-  };
+const Scanner = forwardRef(({
+  styles,
+  containerStyles: { scanner, cameraOverlay, cameraRoll } = {},
+  onQRCodeRead,
+  onClose,
+  isCameraOpen,
+  fullScreen,
+  permissionDialogTitle,
+  permissionDialogMessage,
+  reference,
+  navigation
+}, ref) => {
+  const [camera, setCamera] = useState({
+    permission: 'undetermined',
+    visible: false,
+  });
 
-  setPermissions = permissions => {
-    const { camera, photo } = this.state;
-    camera.permission = permissions.camera;
-    photo.permission = permissions.photo;
-    this.setState({ camera, photo });
-  };
-
-  toggleCamera = () => {
-    const { camera } = this.state;
-    const { isCameraOpen } = this.props;
-
-    camera.visible = !camera.visible;
-    if (typeof isCameraOpen === 'function') {
-      isCameraOpen(camera.visible);
-    }
-    this.setState({ camera });
+  const [photo, setPhoto] = useState({
+    permission: 'undetermined',
+    visible: false,
+  });
 
-    if (!camera.visible && typeof this.props.onClose === 'function') {
-      this.props.onClose();
-    }
+  const setPermissions = permissions => {
+    setCamera(prevState => ({ ...prevState, permissions: permissions.camera }));
+    setPhoto(prevState => ({ ...prevState, permissions: permissions.photo }));
   };
 
-  toggleGallery = () => {
-    const { photo } = this.state;
-    photo.visible = !photo.visible;
-    this.setState({ photo });
+  const checkPermissions = () => {
+    Permissions.checkMultiple(['camera', 'photo']).then(response => {
+      setPermissions(response);
+    });
```

Comment:
as checkPermissions is used in a event listener it should be memorized by useCallback other vise it may refer to a none exciting function
```suggestion
  const checkPermissions = useCallback(() => {
    Permissions.checkMultiple(['camera', 'photo']).then(response => {
      setPermissions(response);
    }), []);
```